src/platforms/dotnet/guides/google-cloud-functions/index.mdx

```diff
@@ -0,0 +1,190 @@
+    "MinimumBreadcrumbLevel": "Debug",
+    "MinimumEventLevel": "Warning",
+    "AttachStackTrace": true,
+    "Debug": true,
+    "DiagnosticsLevel": "Error"
+  }
+```
+
+> An example of some of the options that can be configured via `appsettings.json`.
+
+ASP.NET Core will automatically read this environment variable and bind it to the SDK configuration object.
+
+Some of the settings require actual code. For those, like the `BeforeSend` callback you will need to create a new class implementing the abstract class `FunctionsStartup` to set it up:
+
+```csharp
+using Google.Cloud.Functions.Hosting;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Sentry.AspNetCore;
+
+[assembly: FunctionsStartup(typeof(CustomOptions))]
+
+public class CustomOptions : FunctionsStartup
+{
+    public override void ConfigureServices(WebHostBuilderContext context, IServiceCollection services)
+    {
+        base.ConfigureServices(context, services);
+        services.Configure<SentryAspNetCoreOptions>(options =>
+        {
+            options.BeforeSend = @event =>
+            {
+                @event.ServerName = null;
+                return @event;
+            };
+        });
+    }
+}
+```
+
+> Example modifying all events before they are sent to avoid server names being reported.
+
+## Dependency Injection
+
+Much of the behavior of the Google cloud functions integration with Sentry can be customized by using the frameworks dependency injection system. That is done by registering your own implementation of some of the exposed abstraction inside of `FunctionsStartup`.
+
+### Capturing the Affected User
+
+When opting-in to [SendDefaultPii](#senddefaultpii), the SDK will automatically read the user from the request by inspecting `HttpContext.User`. Default claim values like `NameIdentifier` for the _Id_ will be used.
+
+If you wish to change the behavior of how to read the user from the request, you can register a new `IUserFactory` into the container:
+
+```csharp
+using Google.Cloud.Functions.Hosting;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Sentry.AspNetCore;
+
+[assembly: FunctionsStartup(typeof(UserFactoryStartup))]
+
+public class UserFactoryStartup : FunctionsStartup
+{
+    public override void ConfigureServices(WebHostBuilderContext context, IServiceCollection services)
+    {
+        base.ConfigureServices(context, services);
+        services.AddSingleton<IUserFactory, MyUserFactory>();
+    }
+}
+```
+
+## Options & Initialization
+
+As previously mentioned, this package is a wrapper around [Sentry.Extensions.Logging](/platforms/dotnet/guides/extensions-logging/), [Sentry.AspNetCore](/platforms/dotnet/guides/aspnetcore/) and [Sentry](/platforms/dotnet/). Please refer to the documentation of these packages to get the options that are defined at those levels.
+
+Below, the options that are specific to `Sentry.Google.Cloud.Functions` will be described.
```

Comment:
It says that below are specifics to this package but the values below seems copied from other packages we tell the user to refer to the docs already above